pennylane/qnode.py

```diff
@@ -184,10 +184,25 @@ def __init__(
         self.qfunc_output = None
         # store the user-specified differentiation method
         self.diff_method = diff_method
+        self.diff_method_change = False
 
         self._tape, self.interface, self.device, tape_diff_options = self.get_tape(
             device, interface, diff_method
         )
+        # Update diff_method to reflect which method has been selected
+        if self.diff_method == "best":
+            if tape_diff_options["method"] == "device":
+                self.diff_method = "device"
+                self.diff_method_change = True
+            elif tape_diff_options["method"] == "backprop":
+                self.diff_method = "backprop"
+                self.diff_method_change = True
+            elif tape_diff_options["method"] == "best":
+                self.diff_method = "parameter-shift"
+                self.diff_method_change = True
+            elif tape_diff_options["method"] == "numeric":
+                self.diff_method = "finite-diff"
+                self.diff_method_change = True

```

Comment:
I believe you could factor out the `self.diff_method_change = True` since it will always be called.
```suggestion
            self.diff_method_change = True
            if tape_diff_options["method"] == "device":
                self.diff_method = "device"
            elif tape_diff_options["method"] == "backprop":
                self.diff_method = "backprop"
            elif tape_diff_options["method"] == "best":
                self.diff_method = "parameter-shift"
            elif tape_diff_options["method"] == "numeric":
                self.diff_method = "finite-diff"

```